File: rb/features.rb
# BeverageMixing SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module BeverageMixingFeatures
  def self.make_feature(name)
    case name
    when "base"
      BeverageMixingBaseFeature.new
    when "ratelimit"
      BeverageMixingRatelimitFeature.new
    when "retry"
      BeverageMixingRetryFeature.new
    when "test"
      BeverageMixingTestFeature.new
    when "timeout"
      BeverageMixingTimeoutFeature.new
    else
      BeverageMixingBaseFeature.new
    end
  end
end

File: py/features.py
# BeverageMixing SDK feature factory

from feature.base_feature import BeverageMixingBaseFeature
from feature.test_feature import BeverageMixingTestFeature


def _make_feature(name):
    features = {
        "base": lambda: BeverageMixingBaseFeature(),
        "test": lambda: BeverageMixingTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
